src/pallene/pallenelib.lua

```diff
@@ -493,16 +510,31 @@ static lua_Integer pallene_math_floor(lua_State *L, const char* file, int line,
 /* Based on math_log from lmathlib.c
  * The C compiler should be able to get rid of the if statement if this function is inlined
  * and the base parameter is a compile-time constant */
-static lua_Number pallene_math_log(lua_Integer x, lua_Integer base)
+static lua_Number pallene_math_log(lua_State *L, lua_Number x, TValue optbase)
 {
-    if (base == l_mathop(10.0)) {
-        return l_mathop(log10)(x);
+
+    if (ttisnil(&optbase)) {
+        return l_mathop(log)(x);
+    } else {
+        lua_Number base;
+        if (ttisfloat(&optbase)) {
+            base = fltvalue(&optbase);
+        } else if (ttisinteger(&optbase)) {
+            base = cast(lua_Number, ivalue(&optbase));
+        } else {
+            luaL_error(L, "math log expects a number");
+            PALLENE_UNREACHABLE;
+        }
+
+        if (base == l_mathop(10.0)) {
+            return l_mathop(log10)(x);
 #if !defined(LUA_USE_C89)
-    } else if (base == l_mathop(2.0)) {
-        return l_mathop(log2)(x);
+        } else if (base == l_mathop(2.0)) {
+            return l_mathop(log2)(x);
```

Comment:
Fix indentation

Reply:
Can you please clarify what you mean here? This indentation was intentional. It's similar to what Lua does in lmathlib.c

Reply:
Ifdefs are weird :)